File: Programmers/Level 2/7.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> m;
    map<string, int>:: iterator it;

    for (int i = 0; i < clothes.size(); i++) {
        m[clothes[i][1]]++;
    }

    for (auto c : m) {
        answer *= (c.second + 1);
    }

    answer--;
    
    return answer;
}

File: Programmers/Level-2/7.py
def solution(clothes):
    answer = 1
    hashmap = {} # dictionary
    for cloth in clothes:
        if cloth[1] in hashmap:
            hashmap[cloth[1]] += 1
        else:
            hashmap[cloth[1]] = 2 # 입는 경우(1) + 안 입는 경우(1)
            
    for key in hashmap:
        answer *= hashmap[key]
    
    return answer - 1
